pyanaconda/ui/gui/spokes/storage.py

```diff
@@ -392,81 +364,48 @@ def _method_radio_button_toggled(self, radio_button):
                 self._check_problems()
 
     def apply(self):
-        apply_disk_selection(self._selected_disks)
-        self._auto_part_module.SetEnabled(self._auto_part_enabled)
-
-        self._auto_part_module.SetRequest(
-            PartitioningRequest.to_structure(self._partitioning_request)
-        )
-
-        boot_drive = self._bootloader_module.Drive
-        if boot_drive and boot_drive not in self._selected_disks:
-            reset_bootloader()
-
         self._disk_init_module.SetInitializeLabelsEnabled(True)
-
-        if not self._auto_part_missing_passphrase:
-            self._initialization_mode = CLEAR_PARTITIONS_NONE
-            self._disk_init_module.SetInitializationMode(CLEAR_PARTITIONS_NONE)
+        self._disk_init_module.SetInitializationMode(CLEAR_PARTITIONS_NONE)
+        apply_disk_selection(self._selected_disks, reset_boot_drive=True)
 
     @async_action_nowait
     def execute(self):
+        """"Apply a partitioning."""
+        # Make sure that we apply a non-interactive partitioning.
+        if self._last_partitioning_method == PARTITIONING_METHOD_CUSTOM:
+            log.debug("Skipping the execute method for the CUSTOM partitioning method.")
+            return
+
+        if self._last_partitioning_method == PARTITIONING_METHOD_CUSTOM:
```

Comment:
Shouldn't the check here be different from the one a few lines above ? (Both checking PARTITIONING_METHOD_CUSTOM)

Reply:
Fixed.